librepo/checksum.c

```diff
@@ -248,6 +248,8 @@ lr_checksum_fd_compare(LrChecksumType type,
                     g_debug("%s: Using checksum cached in xattr: [%s] %s",
                             __func__, checksum_key, buf);
                     *matches = (strcmp(expected, buf) == 0);
+                    if (calculated)
+                      *calculated = g_strdup(checksum);
```

Comment:
`checksum` is always NULL here. Did you mean to return the cached checksum value stored in `buf`?

Reply:
You're right: I published this without even building it (for shame). As penance and because it's a good idea, I've both fixed this silly mistake, and extended testing to formalize the use of `lr_checksum_fd_compare()` as a way to calculate or fetch checksums explicitly, which is the point of https://github.com/rpm-software-management/libdnf/pull/1164